### DISCUSSION TITLE
Feature: Mount NFS/SAMBA shares from UI

### DISCUSSION THREAD

**Discussion by Joooost** [IceWhaleTech/CasaOS#168] [Ideas] [answered]
#### Is your idea related to a problem? Please describe:
- In my, and also for friends and colleagues, case I have besides tinkering boards also an existing server or NAS which holds a lot of media files. It would be great to mount NFS/SAMBA shares from the UI. This can be helpful for backups/transferring data and to simple mount media directories which are stored on the NAS so Plex/Sonar/etc can use these locations.

#### Describe the solution you'd like:
- From the UI, somewhere in the disks section, I would love to see an option to add NFS/SAMBA shares to CasaOS. There should be an option to define:
- the server
- path 
- credentials (form SAMBA)
- Option to auto mount on reboot/restart CasaOS
- Option to define (read/write) modus 

#### Describe alternatives you've considered:
- Manually install the necessary tools to mount NFS/SAMBA shares and run script/command from terminal

**Comment by Lauren-ED209** [comment]
Following this line of thinking, should we also consider cloud storage access? Do you think it is valuable or fits with a specific scenario?

**Reply by Joooost** [reply]
Yeah that would be awesome for:
- offsite backups
- map online photo storage
- syncing

Which cloud storage service?
- Amazon S3
- Google (drive)
- OneDrive personal / business?
- WebDav
- ...

**Reply by Lauren-ED209** [reply]
From the request point of view, I think the more the better...WebDav is good for me...

**Reply by sio** [reply]
Rclone may be useful. Adding it will enable support for a lot of storage providers at once

**Reply by Joooost** [reply]
Rclone will help and indeed already supports a lot of  storage providers, still a bit technical to setup/use. Rclone does have an experimental GUI https://rclone.org/gui/, maybe the CasaOS team can look in to this :-)

**Reply by DrMxrcy** [reply]
> Rclone will help and indeed already supports a lot of storage providers, still a bit technical to setup/use. Rclone does have an experimental GUI https://rclone.org/gui/, maybe the CasaOS team can look in to this :-)

I have been updating https://github.com/DrMxrcy/rclonebrowser-docker/ as well :)

**Reply by jerrykuku** [reply]
> > Rclone will help and indeed already supports a lot of storage providers, still a bit technical to setup/use. Rclone does have an experimental GUI https://rclone.org/gui/, maybe the CasaOS team can look in to this :-)
> 
> I have been updating https://github.com/DrMxrcy/rclonebrowser-docker/ as well :)

Cool project！

**Reply by Protektor-Desura** [reply]
If the point of CasaOS is your own cloud home server and keeping control of your own data and improved security by not having a company in the cloud decide what you can store and make you the product, then it would seem that supporting cloud storage options from Dropbox, G-Drive, etc would not be in line with the supposed goals of CasaOS. I would think the point of CasaOS would be to get users used to storing files and backing up their files on their local server they run.

**Comment by JohnGuan** [comment, marked as answer]
Mounting SMB has been implemented in CasaOS [v0.3.5](https://github.com/IceWhaleTech/CasaOS/releases/tag/v0.3.5)
NFS is planned to be implemented in v0.3.6

**Reply by Protektor-Desura** [reply]
Is there going to be a web interface to manage your Samba server and eventually your Samba Active Directory settings & user management?

**Reply by mbox5** [reply]
that would be awesome!

**Comment by Daenks** [comment]
What ever happened to NFS support? Not being able to use my SAN for CasaOS offload storage is really eating away at my use cases.

**Comment by drtrdr23** [comment]
I would like NFS support too.

**Comment by speedrapide10** [comment]
Still no way to mount NFS share to CasaOS v0.4.4.2 🥲

any suggestions ?

**Comment by jan-dewit** [comment]
@JohnGuan can you give an update regarding NFS support?

**Comment by Natetronn** [comment]
I'm hoping for NFS support as well. Is this still in the works?

**Comment by nbcjr** [comment]
How do i mount smb shares using the gui?

**Reply by Natetronn** [reply]
Perhaps this video can help: https://www.youtube.com/watch?v=KA-D60qhNm0

**Comment by colemorgan** [comment]
This is not answered - It was never released.

**Comment by HWiese1980** [comment]
Still no NFS support? Why is this so hard?
